op-node/rollup/finality/finalizer.go

```diff
@@ -236,33 +245,20 @@ func (fi *Finalizer) tryFinalize() {
 	// go through the latest inclusion data, and find the last L2 block that was derived from a finalized L1 block
 	for _, fd := range fi.finalityData {
 		if fd.L2Block.Number > finalizedL2.Number && fd.L1Block.Number <= fi.finalizedL1.Number {
-			// check if fd.L2Block.Number is finalized on Babylon
-			queryParams := &sdk.L2Block{
-				BlockHeight:    fd.L2Block.Number,
-				BlockHash:      fd.L2Block.Hash.String(),
-				BlockTimestamp: fd.L2Block.Time,
+			lastFinalizedBlock, hasErr := fi.findLastFinalizedL2BlockWithConsecutiveQuorom(fd.L2Block.Number, finalizedL2.Number)
+
+			// set finalized block(s)
+			if lastFinalizedBlock != (eth.L2BlockRef{}) {
+				finalizedL2, finalizedDerivedFrom = fi.updateFinalized(lastFinalizedBlock, lastFinalizedBlock.L1Origin)
 			}
-			fi.log.Debug(
-				"babylon gadget query params",
-				"block_height", queryParams.BlockHeight,
-				"block_hash", queryParams.BlockHash,
-				"block_timestamp", queryParams.BlockTimestamp,
-			)
-			babylonFinalized, err := fi.babylonFinalityClient.QueryIsBlockBabylonFinalized(queryParams)
-			fi.log.Debug("babylon gadget query result", "babylon_finalized", babylonFinalized)
-
-			// If the error encountered is of type NoFpHasVotingPowerError, it should be ignored;
-			// for any other error types, emit a critical error event.
-			if err != nil && !errors.Is(err, sdk.ErrNoFpHasVotingPower) {
-				fi.emitter.Emit(rollup.CriticalErrorEvent{Err: fmt.Errorf("failed to check if block %d is finalized on Babylon: %w", fd.L2Block.Number, err)})
+
+			if hasErr {
 				return
```

Comment:
thinking deeper on it. "return on err" is inappropriate now. if you see the below logic where `if finalizedDerivedFrom != (eth.BlockID{}) {` is checked, whenever `finalizedDerivedFrom` is set, we have to do some sanity checks.

so we need to:
- remove the "return on error" logic
- `findLastFinalizedL2BlockWithConsecutiveQuorom` doesn't need to return `hasErr` now

Reply:
fixed in https://github.com/babylonchain/optimism/pull/23/commits/b28870d27649531ca240608a03f48dc8084b0daa